Packages/com.unity.inputsystem/InputSystem/Controls/InputControlPath.cs

```diff
@@ -539,7 +539,7 @@ private static bool StringMatches(Substring str, InternedString matchTo)
             // Can't compare lengths here because str may contain wildcards and
             // thus be shorter than matchTo and still match.
 
-            var matchToLowerCase = matchTo.ToLower();
+            var matchToStringUpper = matchTo.ToString().ToUpperInvariant();
```

Comment:
I'm wondering why InternedString returns str.toString rather than str.m_StringOriginalCase
The later could avoid the GC allocation ?

https://sourcegraph.com/github.com/Unity-Technologies/InputSystem@30277637e5e9986c8cad3be939ad16185ba6114a/-/blob/Packages/com.unity.inputsystem/InputSystem/Utilities/InternedString.cs?L27:19-27:33

public static implicit operator string(InternedString str)

Reply:
But agree switching to uppercase internals in InternedString seems the right thing to do